yt_dlp/extractor/tvw.py

```diff
@@ -0,0 +1,121 @@
+                'description': 'md5:861396cc523c9641d0dce690bc5c35f3',
+                'timestamp': 946389600,
+                'upload_date': '19991228',
+            },
+    }]
+
+    def _get_subtitles(self, response):
+        return {'en': [{'ext': 'vtt', 'url': response.get('captionPath')}]}
+
+    def _get_thumbnail(self, response):
+        return [{'url': response.get('videoThumbnail')}]
+
+    def _get_description(self, response):
+        return clean_html(response.get('description'))
+
+    def _get_js_code(self, video_id, webpage):
+        app_js_url = self._html_search_regex(
+            r'<script[^>]+src=[\"\'](?P<app_js>.+?)[\"\'][^>]+id=\"invintus-app-js\"',
+            webpage, 'app_js')
+        return self._download_webpage(app_js_url, video_id, 'Downloading app.js API key')
+
+    def _extract_formats(self, response, video_id, stream_url):
+        extract_formats = lambda url, video_id: self._extract_m3u8_formats(url, video_id, 'mp4')
+
+        try:
+            return extract_formats(stream_url, video_id)
+        except ExtractorError:
+            self.report_warning('Failed to parse the m3u8 stream. Falling back to the backup stream if it exists.')
+            try:
+                stream_url = traverse_obj(response, ('streamingURIs', 'backup'))
+                return extract_formats(stream_url, video_id)
+            except ExtractorError:
+                raise
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        app_js_code = self._get_js_code(video_id, webpage)
+
+        try:
+            api_key = self._search_regex(r'embedderAuth[\s]*:[\s]*{[^}]+vendorKey:\"(?P<token>\w+?)\"}',
+                                         app_js_code, 'token')
+        except RegexNotFoundError:
+            self.report_warning('Failed to extract the API key. Falling back to a hardcoded key.')
+            api_key = self.BACKUP_API_KEY
+
+        client_id = self._html_search_meta('clientID', webpage)
+        video_id = self._html_search_meta('eventID', webpage)
+
+        try:
+            headers = {'authorization': 'embedder', 'wsc-api-key': api_key}
+            data = json.dumps({'clientID': client_id, 'eventID': video_id,
+                               'showStreams': True}).encode('utf8')
+            response = self._download_json('https://api.v3.invintus.com/v2/Event/getDetailed',
+                                           video_id, headers=headers, data=data).get('data')
+        except ExtractorError as e:
+            if isinstance(e.cause, HTTPError):
+                self.write_debug(e.cause)
+                response = self._parse_json(e.cause.response.read().decode(), video_id).get('errors')
+                if response.get('hasError') is True:
+                    raise ExtractorError('{} said: {}'.format(self.IE_NAME, response.get('error')), expected=True)
+            raise
```

Comment:
I would just use the yt-dlp build in error handling. If there are specific cases you want to catch (geo restrictions, account requirements), you should set the `expected_status` parameter and then throw appropriate error messages based on the response.
```suggestion
        video_data = self._download_json(
            'https://api.v3.invintus.com/v2/Event/getDetailed', video_id,
            headers={
                'authorization': 'embedder',
                'wsc-api-key': '7WhiEBzijpritypp8bqcU7pfU9uicDR',
            },
            data=json.dumps({
                'clientID': client_id,
                'eventID': video_id,
                'showStreams': True,
            }).encode()).get('data')
```